'Obvious' typo __acpp_int8 (should be __acpp_int16) in sscp/atomic_builtins.hpp

**Bug summary**
.../hipSYCL/sycl/libkernel/sscp/atomic_builtins.hpp:41:37: error: size of '__builtin_bit_cast' source type 'unsigned short' does not match destination type '__acpp_int8' (aka 'signed char') (2 vs 1 bytes)
   41 |                                     __builtin_bit_cast(__acpp_int8, x));

**To Reproduce**
instantiate atomic_ref with <uint16_t> within SSCP env

**Expected behavior**
no compilation error

**Describe your setup**
bug still present in AdaptiveCpp develop as of 2025-07-21 commit @650cc8b1 

**Optional additional diagnostic information**
see include/hipSYCL/sycl/libkernel/sscp/atomic_builtins.hpp41:37:

```
template <access::address_space S, class T>
HIPSYCL_BUILTIN void __acpp_atomic_store(T *addr, T x, memory_order order,
                                            memory_scope scope) noexcept {
  if constexpr (sizeof(T) == 1) {
    __acpp_sscp_atomic_store_i8(S, order, scope,
                                   reinterpret_cast<__acpp_int8 *>(addr),
                                   __builtin_bit_cast(__acpp_int8, x));
  } else if constexpr (sizeof(T) == 2) {
    __acpp_sscp_atomic_store_i16(S, order, scope,
                                    reinterpret_cast<__acpp_int16 *>(addr),
                                    __builtin_bit_cast(__acpp_int8, x));
                                                       ^^^^^^^^^^^
```

--- a/include/hipSYCL/sycl/libkernel/sscp/atomic_builtins.hpp
+++ b/include/hipSYCL/sycl/libkernel/sscp/atomic_builtins.hpp
@@ -38,7 +38,7 @@ HIPSYCL_BUILTIN void __acpp_atomic_store(T *addr, T x, memory_order order,
   } else if constexpr (sizeof(T) == 2) {
     __acpp_sscp_atomic_store_i16(S, order, scope,
                                     reinterpret_cast<__acpp_int16 *>(addr),
-                                    __builtin_bit_cast(__acpp_int8, x));
+                                    __builtin_bit_cast(__acpp_int16, x));
   } else if constexpr (sizeof(T) == 4) {
     __acpp_sscp_atomic_store_i32(S, order, scope,
                                     reinterpret_cast<__acpp_int32 *>(addr),

